witness/README.md

```diff
@@ -0,0 +1,44 @@
+Witnessing
+============
+
+Witnessing is an important part of any transparent ecosystem, as it ensures that
+misbehavior on the part of logs can be prevented or detected. This is done by
+ensuring that the _checkpoints_ published by logs are _consistent_, which means 
+the log is presenting a globally consistent view of its contents to everyone and
+thus isn't carrying out a _split-view attack_.
+
+In terms of checkpoints, we consider witness implementations that use the
+[format defined in this
+repository](https://github.com/google/trillian-examples/tree/master/formats/log).
+
+In terms of checking consistency, witnesses can do so in the following two 
+(non-exhaustive) ways:
+- Getting a "classical" consistency proof, assuming all they have is a log root.
+- Getting a compact range, assuming they maintain a compact range.
```

Comment:
"maintain a compact range covering the entirety of the log they've determined to be consistent" (or something, just to avoid the assumption that they get a CR and store the CR as-is)?

Reply:
Cool, updated to hopefully make this more clear.